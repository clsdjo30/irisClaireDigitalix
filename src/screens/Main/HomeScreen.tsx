import React from 'react';
import { StyleSheet, Text, Image, View, Pressable } from 'react-native';
import { getAuth } from 'firebase/auth';
import { useUserStore } from '../../utils/hooks/useUserStore';
import { useUserInformation } from '../../utils/hooks/useUserInformations';
import { colors } from '../../theme';
import { StackScreenProps } from '@react-navigation/stack';
import { LinearGradient } from 'expo-linear-gradient';

const auth = getAuth();

const rightArrow = require('../../../assets/icons/caretRight.png');
const yesNo = require('../../../assets/images/testVector/yes_no.png');
const question = require('../../../assets/images/testVector/simple_question.png');
const questionPlus = require('../../../assets/images/testVector/question_plus.png');

const HomeScreen: React.FC<StackScreenProps<any>> = ({ navigation }) => {


  const userInformation = useUserInformation();

  console.log('userInformation', userInformation);

  const [user, setUser] = useUserStore();

  function goToYesDraw() {
    navigation.navigate('YesDraw');
  }

  function goToCrossDraw() {
    navigation.navigate('CrossDraw');
  }

  function goToDayDraw() {
    navigation.navigate('Tirage');
  }

  return (

    <LinearGradient
      // Card Linear Gradient
      colors={[colors.palette.purple600, colors.palette.purple500]}
      style={styles.container}>
      {/* START 3 ACTIONS CARD */}
      <View style={styles.domainsContainer}>
                
                <Pressable style={styles.domainCard} onPress={goToYesDraw}>
                    <Image source={yesNo} style={styles.icon} />
                    <View style={styles.direction}>
                        <Text style={styles.domainText}>Poser une question Simple</Text>
                        {/* <Image source={rightArrow} style={styles.iconImage}></Image> */}
                    </View>
                </Pressable>
                <Pressable style={styles.domainCard} onPress={goToCrossDraw}>
                        <Image source={questionPlus} style={styles.icon} />
                    <View style={styles.direction}>
                        <Text style={styles.domainText}>Poser une question compl√®te</Text>
                        {/* <Image source={rightArrow} style={styles.iconImage}></Image> */}
                    </View>
                </Pressable>
                <Pressable style={styles.domainCard} onPress={goToDayDraw}>
                        <Image source={question} style={styles.icon} />
                    <View style={styles.direction}>
                        <Text style={styles.domainText}>Tirage du jour</Text>
                        {/* <Image source={rightArrow} style={styles.iconImage}></Image> */}
                        </View>
                </Pressable>
                
            </View>
    </LinearGradient>


  );
}


const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.palette.outterSpace,
    alignItems: 'center',
    justifyContent: 'center',
  },
  // Domain Container
  domainsContainer: {
    width: "100%",
    height: "100%",
    alignItems: "center",
    justifyContent: "center",
  },
  domainCard: {
    width: "90%",
    height: "16%",
    margin: 10,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-around",
    borderWidth: 1,
    borderColor: colors.palette.ivory,
    borderRadius: 10,
    backgroundColor: colors.palette.purple100,
  },
  icon: {
    width: '35%',
    height: '70%',
},
  domainText: {
    width: "90%",
    fontFamily: "mulishBold",
    fontSize: 18,
    color: colors.palette.pink500,
    textAlign: "center",
    marginTop: 20,
  },
  iconImage: {
    width: 30,
    height: 30,
   
  },
  direction: {
    width: "60%",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
  },

});

export default HomeScreen;