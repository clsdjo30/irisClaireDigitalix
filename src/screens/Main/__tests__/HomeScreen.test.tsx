import React from 'react';
import { render, act, cleanup } from '@testing-library/react-native';
import renderer, { ReactTestRenderer } from 'react-test-renderer';

import HomeScreen from '../HomeScreen';

// Mocks
jest.mock('react-native/Libraries/Image/Image', () => 'Image');
jest.mock('../../../hooks/useUserInformations', () => ({
    useUserInformation: () => ({
        user: {
            firstname: 'John',
        },
    }),
}));
jest.mock('react-native-vector-icons/FontAwesome', () => 'Icon');
jest.mock('firebase/auth', () => ({
    getAuth: jest.fn(() => ({
        currentUser: { uid: '123', email: 'test@test.com' },
    })),
}));
jest.mock('../../../utils/resetAtMidnight');

// Mock navigation and route
const mockNavigation: any = {
    navigate: jest.fn(),
};
const mockRoute: any = {
    key: 'MockKey',
    name: 'MockName',
};

describe('HomeScreen', () => {
    afterEach(() => {
        cleanup();
        jest.clearAllMocks();
    });
    
    it('renders correctly', () => {
        let tree: ReactTestRenderer | null = null;
        act(() => {
            tree = renderer.create(<HomeScreen navigation={mockNavigation} route={mockRoute} />);
        });
        if (tree) {
            expect(tree).not.toBeNull();
            expect((tree as ReactTestRenderer).toJSON()).toMatchSnapshot();
        }
    });

    it('should display the greeting', () => {
        const { getByText } = render(<HomeScreen navigation={mockNavigation} route={mockRoute} />);
        expect(getByText('Bonjour John')).toBeTruthy();
    });

    it('should render the CardChoices component', () => {
        const { getByText } = render(<HomeScreen navigation={mockNavigation} route={mockRoute} />);
        expect(getByText('Question Oui/Non')).toBeTruthy();
        expect(getByText('Tirage Complet')).toBeTruthy();
    });

    it('should render the DayCardChoices component with isdraw as false', () => {
        jest.mock('../../../hooks/useDayDrawStore', () => ({
            useDaydrawStore: () => [{ isdraw: false }],
        }));
        const { getByText, getByTestId } = render(<HomeScreen navigation={mockNavigation} route={mockRoute} />);
        expect(getByText('Allez vite découvrir la tendance de votre journée !')).toBeTruthy();
    });
});
