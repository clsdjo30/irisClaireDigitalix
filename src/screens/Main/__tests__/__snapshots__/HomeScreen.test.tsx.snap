// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#eeedf6",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#2F2A5A",
        "borderBottomLeftRadius": 75,
        "borderBottomRightRadius": 75,
        "height": 450,
        "position": "absolute",
        "top": 0,
        "width": 745,
      }
    }
  />
  <Text
    style={
      {
        "color": "#bc9531",
        "fontFamily": "mulishBold",
        "fontSize": 30,
        "marginTop": 20,
        "position": "absolute",
        "textAlign": "center",
        "textTransform": "capitalize",
        "top": 0,
      }
    }
  >
    Bonjour 
    John
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "height": "35%",
        "position": "absolute",
        "top": -40,
        "width": "90%",
      }
    }
  >
    <View
      style={
        {
          "height": "100%",
          "justifyContent": "center",
          "marginTop": 30,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#eeedf6",
            "fontFamily": "mulishRegular",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Allez vite découvrir la tendance de votre journée !
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "position": "absolute",
      }
    }
  >
    <ViewManagerAdapter_ExpoLinearGradient
      colors={
        [
          4292664046,
          4292664046,
        ]
      }
      endPoint={
        [
          1,
          0.5,
        ]
      }
      locations={
        [
          0.1,
          0.9,
        ]
      }
      startPoint={
        [
          0,
          0.5,
        ]
      }
      style={
        {
          "borderRadius": 15,
          "elevation": 5,
          "flexDirection": "row",
          "height": "35%",
          "marginBottom": 35,
          "width": "90%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderRadius": 15,
            "flexDirection": "row",
            "height": "100%",
            "justifyContent": "flex-start",
            "marginBottom": 20,
            "width": "100%",
          }
        }
      >
        <Image
          source={1}
          style={
            {
              "height": "70%",
              "marginLeft": 10,
              "width": "30%",
            }
          }
          testID="card-icon"
        />
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "column",
              "justifyContent": "space-between",
              "width": "70%",
            }
          }
        >
          <Text
            style={
              {
                "color": "#2F2A5A",
                "fontFamily": "mulishBold",
                "fontSize": 20,
                "marginBottom": 10,
                "textAlign": "center",
                "width": "90%",
              }
            }
          >
            Question Oui/Non
          </Text>
          <Text
            style={
              {
                "color": "#2F2A5A",
                "fontFamily": "mulishLight",
                "fontSize": 12,
                "textAlign": "center",
                "width": "90%",
              }
            }
          >
            Recevez une réponse breve à vos questions les plus simple
          </Text>
        </View>
      </View>
    </ViewManagerAdapter_ExpoLinearGradient>
    <ViewManagerAdapter_ExpoLinearGradient
      colors={
        [
          4292664046,
          4292664046,
        ]
      }
      endPoint={
        [
          1,
          0.5,
        ]
      }
      locations={
        [
          0.1,
          0.9,
        ]
      }
      startPoint={
        [
          0,
          0.5,
        ]
      }
      style={
        {
          "borderRadius": 15,
          "elevation": 5,
          "flexDirection": "row",
          "height": "35%",
          "marginBottom": 35,
          "width": "90%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderRadius": 15,
            "flexDirection": "row",
            "height": "100%",
            "justifyContent": "flex-start",
            "marginBottom": 20,
            "width": "100%",
          }
        }
      >
        <Image
          source={1}
          style={
            {
              "height": "70%",
              "marginLeft": 10,
              "width": "30%",
            }
          }
          testID="card-icon"
        />
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "column",
              "justifyContent": "space-between",
              "width": "70%",
            }
          }
        >
          <Text
            style={
              {
                "color": "#2F2A5A",
                "fontFamily": "mulishBold",
                "fontSize": 20,
                "marginBottom": 10,
                "textAlign": "center",
                "width": "90%",
              }
            }
          >
            Tirage Complet
          </Text>
          <Text
            style={
              {
                "color": "#2F2A5A",
                "fontFamily": "mulishLight",
                "fontSize": 12,
                "textAlign": "center",
                "width": "90%",
              }
            }
          >
            Vous avez de grande interogation, vous voulez ....
          </Text>
        </View>
      </View>
    </ViewManagerAdapter_ExpoLinearGradient>
    <ViewManagerAdapter_ExpoLinearGradient
      colors={
        [
          4292664046,
          4292664046,
        ]
      }
      endPoint={
        [
          1,
          0.5,
        ]
      }
      locations={
        [
          0.1,
          0.9,
        ]
      }
      startPoint={
        [
          0,
          0.5,
        ]
      }
      style={
        {
          "borderRadius": 15,
          "elevation": 5,
          "flexDirection": "row",
          "height": "35%",
          "marginBottom": 35,
          "width": "90%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderRadius": 15,
            "flexDirection": "row",
            "height": "100%",
            "justifyContent": "flex-start",
            "marginBottom": 20,
            "width": "100%",
          }
        }
        testID="day-tendance-card"
      >
        <Image
          source={1}
          style={
            {
              "height": "70%",
              "marginLeft": 10,
              "width": "30%",
            }
          }
        />
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "column",
                "justifyContent": "space-between",
                "width": "70%",
              },
            ]
          }
        >
          <Text
            style={
              {
                "color": "#2F2A5A",
                "fontFamily": "mulishBold",
                "fontSize": 20,
                "marginBottom": 10,
                "textAlign": "center",
                "width": "90%",
              }
            }
          >
            Tendance du Jour
          </Text>
          <Text
            style={
              {
                "color": "#2F2A5A",
                "fontFamily": "mulishLight",
                "fontSize": 12,
                "textAlign": "center",
                "width": "90%",
              }
            }
          >
            Une pensée inspirante pour éclairer votre journée !
          </Text>
        </View>
        <View
          style={
            {
              "left": 105,
              "position": "absolute",
              "top": 38,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FF7648",
                "borderRadius": 50,
                "height": 15,
                "justifyContent": "center",
                "width": 15,
              }
            }
          >
            <Icon
              color="#EEEDF6"
              name="exclamation"
              size={10}
              testID="exclamation-icon"
            />
          </View>
        </View>
      </View>
    </ViewManagerAdapter_ExpoLinearGradient>
  </View>
</View>
`;
